sda/cmd/api/api.go

```diff
@@ -174,3 +183,178 @@ func getFiles(c *gin.Context) {
 	// Return response
 	c.JSON(200, files)
 }
+
+func isAdmin() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		token, err := auth.Authenticate(c.Request)
+		if err != nil {
+			log.Debugln("bad token")
+			c.AbortWithStatusJSON(http.StatusUnauthorized, gin.H{"error": err.Error()})
+
+			return
+		}
+		if !slices.Contains(Conf.API.Admins, token.Subject()) {
```

Comment:
I think and identity is the pair of Subject, Issuer that both need to be checked here? (Issuer to key id makes sense to check when getting the token.)

Reply:
The token validation, (issuer and public key) is done in the `auth.Authenticate` call.

Reply:
I don't recall now, but it makes sense to me to support multiple sources (issuers), in which case we'd should check that as part of the identity here as well.

Reply:
Put it in the wish list for the future, this is by no means the end of the work needed to get the API to a functional state.